### PR TITLE
Utilize pn532_i2c from ESPHome dev

### LINKED ISSUE #29: Use Hardware spi
Esphome supports Hardware SPI why not use it?

the only thing you have to do is use the dedicated pins:
```
spi:
  clk_pin: D5
  mosi_pin: D7
  miso_pin: D6
```
then Esphome will use hardware SPI:
```
[12:00:12][C][spi:081]: SPI bus:
[12:00:12][C][spi:082]:   CLK Pin: GPIO14 (Mode: OUTPUT)
[12:00:12][C][spi:083]:   MISO Pin: GPIO12 (Mode: INPUT)
[12:00:12][C][spi:084]:   MOSI Pin: GPIO13 (Mode: OUTPUT)
[12:00:12][C][spi:085]:   Using HW SPI: YES
```
cs_pin can stay D3
buzzer can stay D8
but LED needs a new pin

### LINKED ISSUE #34: D1 Mini won't boot with PN532 power connected
Initially had same issue on a NodeMCU - then realised it was underpowered for the PN532.
Switched to a D1 Mini - but same, it won't boot to ESP Home when the power is connected.
Boots in fine when Power and GND disconnected - but then obviously can't wake the PN532

Any ideas?

Checked Cables and Pins.

### LINKED ISSUE #44: Unstable WiFi connection
The WiFi connection of the esp8266 to the network is unstable.
When connecting the pn532 module and after the setup of the module the WiFi connection and connection to the API stops & hangs often as evidenced by missing pings.

When the module is (physically) connected in HSU mode (or not setup) the connection is stable.

### LINKED ISSUE #56: Scan "transmission" intermittency
I'm still breadboarding this out so have simple and easy reconfiguration. I'm using a NodeMCUv2 running the tagreader.yaml as provided (although the name is rfid_one). Otherwise it is the standard setup.

Plug in the USB and the logs on HA show no scanning. Odd so i connect the serial monitor and see the output start reeling. I am scanning the tags fine. It is the connection to HA that is the problem. So I watch more and notice that it is connecting to HA for about 7 seconds at a time. It is then staying disconnected for 10 to 70 seconds. I'm in the same room as my wifi access point so its not interference or signal loss.

If there is a power saving mode so once a tag is scanned the system wakes up and then sends adding a delay to the process but functioning i'd be happy as it would have worked (albeit slowly).

### PR DESCRIPTION
**Pull request by jesserockz**
This moves to use `pn532_i2c` which is merged into ESPHome `dev` and will be released with `1.16.0` (unknown timeframe on that though)

closes #29 
closes #34
closes #44 
closes #56

### CODE REVIEW

**Comment by balloob** [comment]
Isn't this a duplicate of #52 ?

**Comment by jesserockz** [comment]
> Isn't this a duplicate of #52 ?

Not exactly, #52 also has the NDEF reading by using another branch of ESPHome

File: README.md [hunk 1]
```diff
@@ -25,25 +25,26 @@ The 3D models for the case are [here](STLs).
...
-This enables the PN532 module to communicate with the D1 over SPI, and is required for the modules to work together!
+This enables the PN532 module to communicate with the D1 over I2C, and is required for the modules to work together!
 
 To flash the reader firmware to your D1 Mini you point ESPHome at [tagreader.yaml](tagreader.yaml).  
-**The tag reader requires ESPHome 1.15.2 or later.**
+> :warning: The tag reader requires the *dev* version of ESPHome until `1.16.0` is released with the `pn532_i2c` component.
 
-If you're new to ESPHome, we recommend that you use the [ESPHome Home Assistant add-on](https://esphome.io/guides/getting_started_hassio.html).
+~~If you're new to ESPHome, we recommend that you use the [ESPHome Home Assistant add-on](https://esphome.io/guides/getting_started_hassio.html).~~
+The dev version of ESPHome is availalble after you add the ESPHome addons repo (https://github.com/esphome/hassio) to the Home Assistant Supervisor
```
Comment (balloob) [inline README.md:47]:
```suggestion
The dev version of ESPHome is available after you add the ESPHome add-ons repository (https://github.com/esphome/hassio) to the Home Assistant Supervisor
```

Why cross out the previous sentence. Don't they point at the same repo?

**Review by balloob — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jesserockz) [reply README.md]: No, the documentation doesn't point to the `esphome/hassio` repo because ESPHome is in the community add-ons

**Review by Randyblo7 — APPROVED** [review]

**Pull request merged**